src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -0,0 +1,373 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexReductionTests
+    {
+        // These tests depend on using reflection to access internals of Regex in order to validate
+        // if, when, and how various optimizations are being employed.  As implementation details
+        // change, these tests will need to be updated as well.  Note that Compiled Regexes
+        // null out the _code field being accessed here, so this mechanism won't work to validate
```

Comment:
Made this comment but it vanished: can arrangements be made for compiled regexes to expose this too, albeit only in debug? Then this test could run against them in debug. (I can’t check that code at the moment to see)

Reply:
I considered that, but it's deterministically cleared as part of construction, and I wouldn't want code to take a dependency on it and then find it blows up in release.